servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -405,6 +405,41 @@ public boolean insertDomainTemplate(String domainName, String templateName, Stri
         return true;
     }
 
+    @Override
+    public boolean updateDomainTemplate(String domainName, String templateName, TemplateMetaData templateMetaData) {
+        DomainStruct domainStruct = getDomainStruct(domainName);
+        ArrayList<TemplateMetaData> templateMetalist = new ArrayList<>();
+        if (domainStruct == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "updateDomainTemplate");
+        }
+        if (domainStruct.getTemplates() == null) {
+            domainStruct.setTemplates(new ArrayList<>());
+        }
+        ArrayList<String> templates = domainStruct.getTemplates();
+        if (!templates.contains(templateName)) {
+            templates.add(templateName);
+        }
+
+        if (domainStruct.getTemplateMeta() == null) {
+            domainStruct.setTemplateMeta(new ArrayList<>());
+        }
+        ArrayList<TemplateMetaData> templateMetaList = domainStruct.getTemplateMeta();
+        if (!templateMetaList.isEmpty()) {
+            for (TemplateMetaData meta : templateMetaList) {
+                if (meta.getTemplateName().equals(templateName)) {
+                    meta.setCurrentVersion(templateMetaData.getLatestVersion());
+                } else {
```

Comment:
this is not still correct since if you have 2 template in your list, when comparing the first element, you're just overriding the full list.

https://github.com/yahoo/athenz/pull/1000

I created a new PR based on your PR. Remove the setTemplateNameAndVersion method from your PR and use the implementation of updateDomainTemplate method from that PR.

Reply:
Done.